app/upgrades/v4/upgrades.go

```diff
@@ -251,31 +294,42 @@ func saveCoStakersToStore(
 		collections.BytesKey,
 		codec.CollValue[costktypes.CostakerRewardsTracker](cdc),
 	)
-	dp := costktypes.DefaultParams()
-	totalScore := math.ZeroInt()
-	// we're writing independent key-value
-	// pairs to storage, the order shouldn't affect the final state
-	for addr, val := range coStakers {
-		sdkAddr := sdk.MustAccAddressFromBech32(addr)
-		rt := costktypes.NewCostakerRewardsTracker(
+
+	sdkAddr := sdk.MustAccAddressFromBech32(stakerAddr)
+	addrKey := []byte(sdkAddr)
+
+	// Try to get existing tracker
+	existing, err := rwdTrackers.Get(ctx, addrKey)
+	if err != nil && !errors.Is(err, collections.ErrNotFound) {
+		return nil, err
+	}
+
+	var rt costktypes.CostakerRewardsTracker
+	if errors.Is(err, collections.ErrNotFound) {
+		// Create new tracker
+		rt = costktypes.NewCostakerRewardsTracker(
 			1,
-			val.ActiveSatoshis,
-			val.ActiveBaby,
+			btcAmount,
+			babyAmount,
 			math.ZeroInt(),
 		)
-		rt.UpdateScore(dp.ScoreRatioBtcByBaby)
-		if err := rwdTrackers.Set(ctx, []byte(sdkAddr), rt); err != nil {
-			return err
-		}
-		totalScore = totalScore.Add(rt.TotalScore)
+	} else {
+		// Update existing tracker
+		rt = existing
+		rt.ActiveSatoshis = rt.ActiveSatoshis.Add(btcAmount)
+		rt.ActiveBaby = rt.ActiveBaby.Add(babyAmount)
 	}
 
-	// make sure current rewards is initialized
-	if _, err := k.GetCurrentRewardsInitialized(ctx); err != nil {
-		return err
+	// Update score
+	dp := costktypes.DefaultParams()
```

Comment:
We can have one parameter instance per upgrade handler to avoid allocating so many of them